src/libraries/System.Threading.Tasks.Dataflow/src/Blocks/TransformManyBlock.IAsyncEnumerable.cs

```diff
@@ -0,0 +1,257 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks.Dataflow.Internal;
+
+namespace System.Threading.Tasks.Dataflow
+{
+    public partial class TransformManyBlock<TInput, TOutput>
+    {
+        /// <summary>Initializes the <see cref="TransformManyBlock{TInput,TOutput}"/> with the specified function.</summary>
+        /// <param name="transform">
+        /// The function to invoke with each data element received.  All of the data from the returned <see cref="System.Collections.Generic.IAsyncEnumerable{TOutput}"/>
+        /// will be made available as output from this <see cref="TransformManyBlock{TInput,TOutput}"/>.
+        /// </param>
+        /// <exception cref="System.ArgumentNullException">The <paramref name="transform"/> is null (Nothing in Visual Basic).</exception>
+        public TransformManyBlock(Func<TInput, IAsyncEnumerable<TOutput>> transform) :
+            this(transform, ExecutionDataflowBlockOptions.Default)
+        { }
+
+        /// <summary>Initializes the <see cref="TransformManyBlock{TInput,TOutput}"/> with the specified function and <see cref="ExecutionDataflowBlockOptions"/>.</summary>
+        /// <param name="transform">
+        /// The function to invoke with each data element received.  All of the data from the returned in the <see cref="System.Collections.Generic.IAsyncEnumerable{TOutput}"/>
+        /// will be made available as output from this <see cref="TransformManyBlock{TInput,TOutput}"/>.
+        /// </param>
+        /// <param name="dataflowBlockOptions">The options with which to configure this <see cref="TransformManyBlock{TInput,TOutput}"/>.</param>
+        /// <exception cref="System.ArgumentNullException">The <paramref name="transform"/> is null (Nothing in Visual Basic).</exception>
+        /// <exception cref="System.ArgumentNullException">The <paramref name="dataflowBlockOptions"/> is null (Nothing in Visual Basic).</exception>
+        public TransformManyBlock(Func<TInput, IAsyncEnumerable<TOutput>> transform, ExecutionDataflowBlockOptions dataflowBlockOptions)
+        {
+            // Validate arguments.
+            if (transform == null) throw new ArgumentNullException(nameof(transform));
+            Initialize(messageId => ProcessMessage(transform, messageId), dataflowBlockOptions, ref _source!, ref _target!, ref _reorderingBuffer, async: true);
+        }
+
+        /// <summary>Processes the message with a user-provided transform function that returns an async enumerable.</summary>
+        /// <param name="transformFunction">The transform function to use to process the message.</param>
+        /// <param name="messageWithId">The message to be processed.</param>
+        private async void ProcessMessage(Func<TInput, IAsyncEnumerable<TOutput>> transformFunction, KeyValuePair<TInput, long> messageWithId)
```

Comment:
is "async **void**" really needed here?

```suggestion
        private async Task ProcessMessage(Func<TInput, IAsyncEnumerable<TOutput>> transformFunction, KeyValuePair<TInput, long> messageWithId)
```

Reply:
`async` keyword removed.

Reply:
But I am not pleased with the usage of `GetAwaiter().GetResult()`...
Is there a better solution here?

Reply:
I think we need sync overloads...

Some workaround would be also creating a method returning a ValueTask and calling .GetAwaiter().GetResult() on it

Reply:
> I think we need sync overloads

Sync overloads of what?